test/lib/commands/query.js

```diff
@@ -0,0 +1,168 @@
+const t = require('tap')
+const { load: loadMockNpm } = require('../../fixtures/mock-npm')
+
+t.cleanSnapshot = (str) => {
+  const normalizePath = p => p
+    .replace(/\\+/g, '/')
+    .replace(/\r\n/g, '\n')
+  return normalizePath(str)
+    .replace(new RegExp(normalizePath(process.cwd()), 'g'), '{CWD}')
+    // normalize between windows and posix
+    .replace(new RegExp('lib/node_modules', 'g'), 'node_modules')
+}
+
+t.test('simple query', async t => {
+  const { npm, joinedOutput } = await loadMockNpm(t, {
+    prefixDir: {
+      node_modules: {
+        a: {
+          name: 'a',
+          version: '1.0.0',
+        },
+        b: {
+          name: 'b',
+          version: '^2.0.0',
+        },
+      },
+      'package.json': JSON.stringify({
+        name: 'project',
+        dependencies: {
+          a: '^1.0.0',
+          b: '^1.0.0',
+        },
+      }),
+    },
+  })
+  await npm.exec('query', [':root, :root > *'])
+  t.matchSnapshot(joinedOutput(), 'should return root object')
```

Comment:
the name of this assertion should probably be `should return root object and direct children` instead

Reply:
👍